### PR TITLE
Add REST implementation for Validator's `GetDuties`

### PR DESCRIPTION
**Pull request by PatriceVignola**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

It adds the `GetDuties` REST API endpoint for the validator.

### CODE REVIEW

File: validator/client/beacon-api/duties.go
Comment (rkapka) [inline validator/client/beacon-api/duties.go:48]: Actually we do: https://github.com/prysmaticlabs/prysm/blob/9f44d6e45292315f6103159d2ef88d9c0c900ea0/beacon-chain/rpc/eth/validator/validator.go#L153

File: validator/client/beacon-api/duties.go
Comment (rkapka) [inline validator/client/beacon-api/duties.go:263]: You may want to check if `proposerDuties.Data == nil` first, like you do in `GetCommittees`.

File: validator/client/beacon-api/duties.go
Comment (rkapka) [inline validator/client/beacon-api/duties.go:289]: You may want to check if `syncDuties.Data == nil` first, like you do in `GetCommittees`.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: validator/client/beacon-api/duties.go
Comment (james-prysm) [inline validator/client/beacon-api/duties.go:209]: I want to double check if using the head in this case is correct, the grpc one read a little differently.

Comment (PatriceVignola) [reply validator/client/beacon-api/duties.go:209]: The way I understand it from the spec, the `epoch` query parameter decides which epoch to query the committees for. The `head` committees will only be fetches if `epoch` isn't specified.

epoch | Fetch committees for the given epoch. If not present then the committees for the epoch of the state will be obtained.
-- | --

Comment (james-prysm) [reply validator/client/beacon-api/duties.go:205]: got it this makes sense.

**Review by james-prysm — APPROVED** [review]
will wait for radek for a second approval.

**Review by rkapka — APPROVED** [review]

**Pull request merged**
